python/ray/autoscaler/v2/autoscaler.py

```diff
@@ -133,16 +138,27 @@ def _init_instance_manager(
         subscribers.append(
             RayStopper(gcs_client=gcs_client, error_queue=self._ray_stop_errors_queue)
         )
-        if not config.disable_node_updaters():
-            # Supporting ray installer is only needed for providers that doesn't
-            # install or manage ray (e.g. AWS, GCP). These providers will be
-            # supported in the future.
-            raise NotImplementedError(
-                "RayInstaller is not supported yet in current "
-                "release of the Autoscaler V2. Therefore, providers "
-                "that update nodes (with `disable_node_updaters` set to True) "
-                "are not supported yet. Only KubeRay is supported for now which sets "
-                "disable_node_updaters to True in provider's config."
+        if not config.disable_node_updaters() and isinstance(
+            cloud_provider, NodeProviderAdapter
+        ):
+            head_node_ip = urlsplit("//" + self._gcs_client.address).hostname
+            assert head_node_ip is not None, "Invalid GCS address format"
+            subscribers.append(
+                ThreadedRayInstaller(
+                    head_node_ip=head_node_ip,
+                    instance_storage=instance_storage,
+                    ray_installer=RayInstaller(
+                        provider=cloud_provider.v1_provider,
+                        config=config,
+                    ),
+                    error_queue=self._ray_install_errors_queue,
+                    # TODO(rueian): Rewrite the ThreadedRayInstaller and its underlying
+                    # NodeUpdater and CommandRunner to use the asyncio, so that we don't
+                    # need to use so many threads. We use so many threads now because
+                    # they are blocking and letting the new cloud machines to wait for
+                    # previous machines to finish installing Ray is quite inefficient.
+                    max_concurrent_installs=config.get_max_num_worker_nodes() or 50,
```

Comment:
The refactor plan for using `asyncio` could be:

1. Introduce `AsyncCommandRunner`s one by one and let the old `CommandRunner`s use the new `AsyncCommandRunner`s under the hood with its own event loop.
2. Introduce `AsyncNodeUpdater` and let the `NodeUpdater` use the `AsyncNodeUpdater` under the hood with its own event loop.
3. Rewrite `ThreadedRayInstaller` to use `AsyncNodeUpdater` and `AsyncCommandRunner`.
4. Remove `NodeUpdater` and `CommandRunner` when autoscaler v1 is deleted.